pyfr/plugins/ascent.py

```diff
@@ -0,0 +1,430 @@
+                    RTLD_GLOBAL)
+import re
+
+import numpy as np
+
+from pyfr.ctypesutil import LibWrapper
+from pyfr.mpiutil import get_comm_rank_root
+from pyfr.nputil import npeval
+from pyfr.plugins.base import BasePlugin, RegionMixin
+from pyfr.shapes import BaseShape
+from pyfr.util import file_path_gen, subclass_where
+from pyfr.writers.vtk import BaseShapeSubDiv
+
+
+class AscentError(Exception): pass
+class ConduitError(Exception): pass
+
+
+class ConduitWrappers(LibWrapper):
+    _libname = 'conduit'
+    _mode = RTLD_GLOBAL
+
+    # Functions
+    _functions = [
+        (c_void_p, 'conduit_node_append', c_void_p),
+        (c_void_p, 'conduit_node_create', c_void_p),
+        (None, 'conduit_node_destroy', c_void_p),
+        (c_void_p, 'conduit_node_fetch', c_void_p, c_char_p),
+        (c_void_p, 'conduit_node_set_path_char8_str', c_void_p, c_char_p,
+         c_char_p),
+        (c_void_p, 'conduit_node_set_path_float32', c_void_p, c_char_p,
+         c_float),
+        (c_void_p, 'conduit_node_set_path_float32_ptr', c_void_p, c_char_p,
+         c_void_p, c_longlong),
+        (c_void_p, 'conduit_node_set_path_float64', c_void_p, c_char_p, 
+         c_double),
+        (c_void_p, 'conduit_node_set_path_float64_ptr', c_void_p, c_char_p,
+         c_void_p, c_longlong),
+        (c_void_p, 'conduit_node_set_path_int64', c_void_p, c_char_p,
+         c_longlong),
+        (c_void_p, 'conduit_node_set_path_int64_ptr', c_void_p, c_char_p,
+         c_void_p, c_longlong),
+        (c_void_p, 'conduit_node_set_path_node', c_void_p, c_char_p, c_void_p),
+        (c_void_p, 'conduit_node_print', c_void_p),
+    ]
+
+    def _errcheck(self, status, fn, args):
+        if not status:
+            raise ConduitError
+
+
+class ConduitNode:
+    def __init__(self, lib, ptr=None, child=False):
+        self.lib = lib
+        self.child = child
+        self._as_parameter_ = ptr or self.lib.conduit_node_create(None)
+
+    def __del__(self):
+        if not self.child:
+            self.lib.conduit_node_destroy(self)
+
+    def __setitem__(self, key, value):
+        key = key.encode()
+        match value:
+            case str():
+                self.lib.conduit_node_set_path_char8_str(self, key, 
+                                                         value.encode())
+            case ConduitNode():
+                self.lib.conduit_node_set_path_node(self, key, value)
+            case int():
+                self.lib.conduit_node_set_path_int64(self, key, value)
+            case float():
+                self.lib.conduit_node_set_path_float32(self, key, value)
+            case (np.ndarray() | np.generic()):
+                fn = getattr(self.lib, 
+                             f'conduit_node_set_path_{value.dtype}_ptr')
+                fn(self, key, value.ctypes.data, value.size)
+            case list():
+                value = np.ascontiguousarray(np.array(value, dtype=np.float64))
```

Comment:
Just an np.array will be fine here.  Also, no need for the getattr as you know what is needed.